src/db.ts

```diff
@@ -352,56 +313,50 @@ export async function fetchLatestDelegationsBlockNumber() {
 
 export async function fetchLatestTransactionsBlockNumber(beforeDate?: Date) {
   let sql = `SELECT block FROM ${TRANSACTIONS_TABLE}`;
+  const params: any = {};
   if (beforeDate) {
-    sql += ` WHERE UNIX_TIMESTAMP(transaction_date) <= ${
-      beforeDate.getTime() / 1000
-    }`;
+    sql += ` WHERE UNIX_TIMESTAMP(transaction_date) <= :date`;
+    params.date = beforeDate.getTime() / 1000;
   }
   sql += ` order by block desc limit 1;`;
-  const r = await execSQL(sql);
+  const r = await sqlExecutor.execute(sql, params);
   return r.length > 0 ? r[0].block : 0;
 }
 
-export async function fetchLatestTDHBDate() {
+export async function fetchLatestTDHBDate(): Promise<Date> {
   const sql = `SELECT timestamp FROM ${TDH_BLOCKS_TABLE} order by block_number desc limit 1;`;
-  const r = await execSQL(sql);
-  return r.length > 0 ? r[0].timestamp : 0;
-}
-
-export async function fetchTdhReplayTimestamp(date: Date) {
-  const sql = `SELECT timestamp FROM ${TDH_BLOCKS_TABLE} WHERE created_at > '2023-03-06' AND timestamp <= ${mysql.escape(
-    date
-  )} order by block_number asc limit 1;`;
-  const r = await execSQL(sql);
-  return r.length > 0 ? r[0].timestamp : null;
+  const r = await sqlExecutor.execute(sql);
+  return r.length > 0 ? new Date(r[0].timestamp) : new Date();
```

Comment:
Are you sure that in case there is nothing, the timestamp should be current time and not epoch for example?

Reply:
will be fixing this one good catch!